### PR TITLE
Antag RIG buff

### PR DESCRIPTION
**Pull request by soslow4u**
- New item: **RIG Autodoc module**
Light version of beloved machine that would fix your traitorous ass when whole ship is against you.
Costs **16 TC**, you think it worth it? Let me know if this is too much.
- Added basic chemical dispenser for uplink. Just **8 TC** for 80 units of cool chems for all your needs.
   - Tricordrazine
   - Tramadol
   - Dexalin+
   - Spaceacillin
   - Dylovene and many more!

### CODE REVIEW

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 1]
```diff
@@ -380,4 +377,102 @@
...
+	if(!.)
+		return
+	to_chat(holder.wearer, SPAN_NOTICE("With quiet hum internal machinery starts checking your body..."))
+	var/mob/living/carbon/human/H = holder.wearer
+
+	for(var/obj/item/organ/external/E in H.bad_external_organs)
+		to_chat(holder.wearer, "Found wounded organ: "+SPAN_WARNING("[E]"))
+		var/datum/rig_surgery/patchnote = new()
+		for(var/datum/wound/internal_bleeding/IB in E.wounds)
+			patchnote.surgery_type += DR_RIG_IB
```
Comment (LethalGhost) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]: That's bit value. 
```suggestion
			patchnote.surgery_type |= DR_RIG_IB
```

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 2]
```diff
@@ -380,4 +377,102 @@
...
+	if(!.)
+		return
+	surgery_list.Cut()
+
+/obj/item/rig_module/autodoc/Process()
+	. = ..()
+	for(var/datum/rig_surgery/patchnote in surgery_list)
+		var/obj/item/organ/external/ex_organ = patchnote.ex_organ
+		if(!ex_organ)
+			CRASH()
```
Comment (LethalGhost) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]: How about `surgery_list -= patchnote` to prevent continuous crashing?

File: code/modules/clothing/spacesuits/rig/modules/utility.dm [hunk 3]
```diff
@@ -380,4 +377,102 @@
...
+		if(patchnote.surgery_type & DR_RIG_IB)
+			for(var/datum/wound/IB in ex_organ.wounds)
+				if(prob(50) && IB.internal)
+					ex_organ.wounds -= IB
```
Comment (LethalGhost) [inline code/modules/clothing/spacesuits/rig/modules/utility.dm]: Why not use `IB.heal_damage(SOME_VALUE)`?

**Comment by LethalGhost** [comment]
В целом.
По моему, стоит заменить 50% шанс на мгновенное лечение каждой раны.
Сделать, например так - в худе рига есть кнопка открывающая интерфейс модуля.
В нем кнопка "сканировать" (или сканирование происходит автоматически).
После сканирования мы получаем список ран носителя рига.
Возле каждого элемента списка кнопка "лечить". Лечение происходит не мгновенно (как и операции) и, возможно, требует носителя стоять неподвижно некоторое время.

Дискас.

ENG:
How about replace 50% chance of immidiatly healing (for each wound) with 
- Special module UI which display all your wounds
- You can heal each wound by clicking "heal" button near it in module UI
- Healing take time (like surgery) and may be require RIG wearer to stay still

Let the discussion begin!

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 2 above)
Comment (soslow4u) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm]: Well, it was late night brainfart about gibbed organs in action, it honestly should just remove patchnote at all.

File: code/modules/clothing/spacesuits/rig/modules/utility.dm (hunk 3 above)
Comment (soslow4u) [reply code/modules/clothing/spacesuits/rig/modules/utility.dm]: Because of `prob()` proc

**Comment by soslow4u** [comment]
1. > Immediately healing.

What if organ has just too much brute/burn value? Is it wise to spend just 500 units of power to get rejuvenate analog? Shrapnel is still a threat, inner organ damage should be too.

2. > RIG UI with each organ.

Another minigame to get your wounds healed under action? I find it not wise to make player click additional buttons when he is getting hit.
But maybe yes, just need to make sure it would be even used by antags

3. > Need to stand still.

Yeah, I had same idea, but then decided to discuss it there. By the way, low healing values/rng rates were set in sake for keeping module active in fight for short amount of time.

tl;dr: need more ideas/screams/discussion about sanity of such suggestion

**Comment by LethalGhost** [comment]
>Another minigame to get your wounds healed under action? I find it not wise to make player click additional buttons when he is getting hit.

Right now it can heal you extremely fast even mend fractures. There is no reason to not use it within a battle.

UI will make it more controllable and balanced - it'll force you to stop for healing. At least for complex operations. We will be able to set different powercost/time for different operations.

That change will force user to not "hope for a luck" but use that tool wisely.

**Comment by soslow4u** [comment]
Well then, to be done.
